File: HackerRank/Javascript/plusMinus.js
// https://www.hackerrank.com/challenges/plus-minus/problem?h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen
// Reprak11

function plusMinus(arr){
    let cnt=[0, 0, 0];
    for (const i in arr){
        (arr[i] > 0) ? cnt[0]++ : (arr[i] < 0) ? cnt[1]++ : cnt[2]++
    }
    for (const i in cnt){
        console.log((cnt[i]/arr.length).toFixed(6))
    }
}


const n = parseInt(prompt(),10);
const arr = prompt().split(' ').map(arrTemp => parseInt(arrTemp,10));
plusMinus(arr);

File: HackerRank/Python/plusMinus.py
# https://www.hackerrank.com/challenges/plus-minus/problem?h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen&h_r=next-challenge&h_v=zen
# Reprak11

def plusMinus(arr):
    cnt = [0, 0, 0]
    for i in range(len(arr)):
        if(arr[i] > 0):
            cnt[0] += 1
        elif (arr[i] < 0):
            cnt[1] += 1
        else:
            cnt[2] += 1
    for element in cnt:
        print("{:.6f}".format(element/len(arr)))  

    return 


if __name__ == '__main__':
    n = int(input())
    arr = list(map(int, input().rstrip().split()))
    plusMinus(arr)